app/classes/Transvision/Utils.php

```diff
@@ -119,8 +119,9 @@ public static function markString($needle, $haystack)
      */
     public static function highlightString($str)
     {
+        // See http://stackoverflow.com/a/14952740 for the regex background
         $str = preg_replace(
-            '/←(.*)→/isU',
+            '/←((?:[^←→]|(?R))*)→/isu',
```

Comment:
On second thoughts we should be able to drop also the 's' [modifier](http://php.net/manual/en/reference.pcre.pattern.modifiers.php). And probably worth adding a better explanation than a link to SO.